### PR TITLE
[HttpFoundation] Avoid implicit null to array conversion in request matcher

### PR DESCRIPTION
**Pull request by ro0NL**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any |
| License | MIT |
| Doc PR | reference to the documentation PR, if any |

`null` is allowed _and_ passed as default value from the constructor. Lets be explicit.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/RequestMatcher.php [hunk 1]
```diff
@@ -76,13 +76,13 @@ public function __construct($path = null, $host = null, $methods = null, $ips =
      */
     public function matchScheme($scheme)
     {
-        $this->schemes = array_map('strtolower', (array) $scheme);
+        $this->schemes = null !== $scheme ? array_map('strtolower', (array) $scheme) : array();
```
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/RequestMatcher.php]: Not really needed though. Confused with `(array) null` actually converting to an empty array, not something like `array(null)` :)

**Comment by Tobion** [comment]
👍

**Comment by javiereguiluz** [comment]
:+1:

(failing tests are unrelated)

**Comment by fabpot** [comment]
Good catch, thanks @ro0NL.

**Pull request closed without merging**